clustering/kubernetes.py

```diff
@@ -357,25 +357,27 @@ def main():
 
     for item in data:
         namespace = "default"
+        parsed_item = {}
         if item and 'metadata' in item:
-            namespace = item.get('metadata', {}).get('namespace', "default")
-            kind = item.get('kind', '').lower()
+            parsed_item = yaml.load(item)
```

Comment:
Since above there is an explicit yaml.load_all when file_reference is used, I think it would be better to only attempt to yaml.load the info when inline_data is specified..., so I would push this up under `if inline_data:` rather than applying this here.

The other thing that I would suggest is to only attempt to do a yaml.load of the data `if not isinstance(dict, inline_data) and not isinstance(list, inline_data):`, so I would do suggest something like this around line 332:

```
if inline_data:
    if not isinstance(dict, inline_data) and not isinstance(list, inline_data):
        data = yaml.load(inline_data)
    else:
        data = inline_data